### DISCUSSION TITLE
Converting a String to JSONContent for Initial Value in Novel Editor

### DISCUSSION THREAD

**Discussion by rossanodr** [steven-tey/novel#454] [Q&A]
Currently, the Novel Editor only accepts JSONContent as the input for initialContent. How can I convert a string into the required JSONContent format to set it as the initial value?

**Comment by kamyarkaviani** [comment]
The issue is that Novel includes the extension [`tiptap-markdown`](https://github.com/steven-tey/novel/blob/ecdb5d2ff5d3e6ab5f5af95b5a81de3ab4e8a0e9/packages/headless/src/extensions/index.ts#L13) and specifically configures it with [`html: false`](https://github.com/steven-tey/novel/blob/ecdb5d2ff5d3e6ab5f5af95b5a81de3ab4e8a0e9/packages/headless/src/extensions/index.ts#L40). This causes markdown extension to escape out any HTML you try to insert.

The solution would be to either include Novel from source and manually disable markdown, or convert your html content to JSON as an additional step outlined here https://tiptap.dev/docs/editor/api/utilities/html.
